core/services/workflows/engine.go

```diff
@@ -678,15 +678,15 @@ func (e *Engine) finishExecution(ctx context.Context, cma custmsg.MessageEmitter
 	executionDuration := int64(execState.FinishedAt.Sub(*execState.CreatedAt).Seconds())
 	switch status {
 	case store.StatusCompleted:
-		e.metrics.UpdateWorkflowCompletedDurationHistogram(ctx, executionDuration)
+		e.metrics.With(platform.KeyWorkflowExecutionID, executionID).UpdateWorkflowCompletedDurationHistogram(ctx, executionDuration)
```

Comment:
@mchain0 You're adapting metrics on the v1 engine here :) 

Also adding high cardinality tags to metrics is explicitly discouraged because it significantly increases cost